lib/ansible/cli/__init__.py

```
@@ -254,6 +254,10 @@ def setup_vault_secrets(loader, vault_ids, vault_password_files=None,
 
         last_exception = found_vault_secret = None
         for vault_id_slug in vault_ids:
+            if not vault_id_slug:
+                # silently ignore empty values
+                continue
+
```

I am not sure we want to silently ignore empty IDs, I do agree they should not be used to 'find a file' but that can be short circuited at that bridge.

---

Could you please elaborate what exactly you mean? To me "that bridge" is the call to `get_file_vault_secret()` which comes 3 LoC later (ignoring the if which is always false in this case), so this should be the proper place to shortcircuit. I also specifically check the whole "slug" and not just the (after @) file part, so something like `foo@` still properly errors.

---

yes, i mean moving the 'empty' check to lines 291/295

---

Done, although I have to admit I fail to see the improvement.

---

its not always a file name, but a secret, sadly i've found some cases of 'empty secret use', which i should really add tests for if we commit to supporting them ...

---

If you can give me an example or a pointer, I can try to cook something up. It's not necessarily about "supporting it", but rather "it works now; if it breaks, make sure it's intentional".